### ISSUE TITLE
npm view deals badly when `latest` points to nonexistent version

### ISSUE THREAD

**Issue by chrisirhc** [npm/npm#11852]
I'm not sure if this is a npm registry corruption issue and if it is, whether this issue belongs here. Please let me know if it goes elsewhere.

I'm getting no information on the react-router package [although it's clearly an existing package](https://www.npmjs.com/package/react-router) with plenty of packages depending on it.

On npm 3.8.0 (also tested on npm 2.14.2 with similar output, no info):

```
➜  test-npm3 ./node_modules/.bin/npm version
{ npm: '3.8.0',
  ares: '1.9.0-DEV',
  http_parser: '1.0',
  modules: '11',
  node: '0.10.32',
  openssl: '1.0.1i',
  uv: '0.10.28',
  v8: '3.14.5.9',
  zlib: '1.2.3' }
➜  test-npm3 ./node_modules/.bin/npm --verbose view react-router
npm info it worked if it ends with ok
npm verb cli [ 'node',
npm verb cli   '/Users/chris/Documents/hacking/test-npm3/node_modules/.bin/npm',
npm verb cli   '--verbose',
npm verb cli   'view',
npm verb cli   'react-router' ]
npm info using npm@3.8.0
npm info using node@v0.10.32
npm verb request uri https://registry.npmjs.org/react-router
npm verb request no auth needed
npm info attempt registry request try #1 at 11:04:39
npm verb request id 9ed89ab55f8b32c1
npm verb etag "1P1GZ0USD4A9GLLX050FIZ0IA"
npm http request GET https://registry.npmjs.org/react-router
npm http 304 https://registry.npmjs.org/react-router
npm verb headers { date: 'Thu, 10 Mar 2016 19:04:42 GMT',
npm verb headers   via: '1.1 varnish',
npm verb headers   'cache-control': 'max-age=300',
npm verb headers   etag: '"1P1GZ0USD4A9GLLX050FIZ0IA"',
npm verb headers   age: '20',
npm verb headers   connection: 'keep-alive',
npm verb headers   'x-served-by': 'cache-sjc3129-SJC',
npm verb headers   'x-cache': 'HIT',
npm verb headers   'x-cache-hits': '2',
npm verb headers   'x-timer': 'S1457636682.381588,VS0,VE0',
npm verb headers   vary: 'Accept' }
npm verb etag https://registry.npmjs.org/react-router from cache
npm verb get saving react-router to /Users/chris/.npm/registry.npmjs.org/react-router/.cache.json
npm verb correctMkdir /Users/chris/.npm correctMkdir not in flight; initializing
npm verb exit [ 0, true ]
npm info ok
➜  test-npm3
```

**Comment by joeframbach** [comment]
`npm outdated` similarly skips over the output for `react-router` entirely. I've also opened an issue on react-router. https://github.com/reactjs/react-router/issues/3170

**Comment by othiym23** [comment]
It's going to take some digging into the code to understand why this is happening, because the data, as far as I can tell, is correct on the registry (you can see it if you run `curl https://registry.npmjs.org/react-router | json`). Either way, confirmed. Thanks for the report!

**Comment by othiym23** [comment]
> Does Github have a good way to handle parallel conversations like these two issues?

No, but now you can add reactions to comments, which is _definitely_ very useful.

**Comment by othiym23** [comment]
@iarna took a look at this, and it appears that somebody set `latest` dist-tag for `react-router` to point to a nonexistent version. The CLI should handle this more gracefully, and the registry shouldn't let you set a dist-tag to point to versions that don't exist, so this one issue has identified at least two bugs. Thanks for pointing this out!

**Comment by joeframbach** [comment]
Anyway, here's what I can tell:

react-router published `2.0.0-rc1`,
then published `2.0.0-rc2`,
then published `2.0.0-rc3`,
then published `2.0.9-rc4`,
then unpublished `2.0.9-rc4`,
then published `2.0.0-rc4`

If you look at the json from `https://registry.npmjs.org/react-router` you can see that `2.0.9-rc4` is in the `time` list, but no such key exists in `versions`.

**Comment by othiym23** [comment]
@ryanflorence has fixed this issue for `react-router`, but I've retitled this issue to reflect the underlying problem so we can track / fix it in the CLI.
